src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -1362,11 +1362,29 @@ public int LastIndexOf(string source, string value, int startIndex, int count, C
 
         internal static int LastIndexOfOrdinal(string source, string value, int startIndex, int count, bool ignoreCase)
         {
+            Debug.Assert(source != null);
+            Debug.Assert(value != null);
+
             if (GlobalizationMode.Invariant)
             {
                 return InvariantLastIndexOf(source, value, startIndex, count, ignoreCase);
             }
 
+            // For ordinal (non-linguistic) comparisons, an empty target string is always
+            // found at the end of the search space, and a non-empty target string
+            // can never be found within an empty search space. This assumption is not
+            // valid for linguistic comparisons, including InvariantCulture comparisons.
+
+            if (value.Length == 0)
+            {
+                return Math.Max(0, startIndex - count + 1);
```

Comment:
This line is because the behavior of `LastIndexOf` is nutty. See the comment below (lifted from PR https://github.com/dotnet/runtime/pull/1514) for more info.

https://github.com/dotnet/runtime/blob/52aea3b2f739682b75f27d5c264f5d2bc89adfe7/src/libraries/System.Private.CoreLib/src/System/String.Searching.cs#L277-L298